### PR TITLE
fcitx5-nord: init at unstable-2021-07-27, fcitx5-material-color: init at 0.2.1

### PR DESCRIPTION
**Pull request by Cryolitia**
## Description of changes

Add two fcitx theme, both of them are already in archlinux extra repo, so I hope that it could help some people ^_^.

https://github.com/hosxy/Fcitx5-Material-Color

https://github.com/tonyfettes/fcitx5-nord

- [x] To confirm `fcitx5-material-color` working

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:35]: Here should end with a new line, see <https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3>

If you're using vscode could setup follow <https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files>

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:30]: Maybe use a more brief description? I'm not sure if it could contain CJK characters or emoji, it may not display correctly with some users.

About requirements: <https://nixos.org/manual/nixpkgs/stable/#var-meta-description>

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:14]: Could use `hash` instead as well.

File: pkgs/by-name/fc/fcitx5-nord/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-nord/package.nix:8]: `version` here should be `unstable-2021-07-27`

<https://nixos.org/manual/nixpkgs/stable/#sec-package-naming>
_If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format._

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:6]: Here is a little suggestion, to use `finalAttrs` pattern instead.

+ [Historical Discussion](https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293)
+ [Documentation](https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes)
+ [Implementation Example](https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe)

File: pkgs/by-name/fc/fcitx5-nord/package.nix
Comment (oluceps) [inline pkgs/by-name/fc/fcitx5-nord/package.nix:6]: Here as well.

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (Cryolitia) [reply pkgs/by-name/fc/fcitx5-material-color/package.nix]: learned new things, thx

**Comment by oluceps** [comment]
Result of `nixpkgs-review pr 263319` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fcitx5-material-color</li>
    <li>fcitx5-nord</li>
  </ul>
</details>

**Review by oluceps — APPROVED** [review]

**Comment by h7x4** [comment]
I don't mind this being the PR that gets merged, I currently also have #263035 up

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:32]: `apsl20` is the Apple Public Source License 2.0, not the Apache one

```suggestion
    license = licenses.asl20;
```

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:13]:
```suggestion
    rev = finalAttrs.version;
```

**Comment by Cryolitia** [comment]
> I don't mind this being the PR that gets merged, but please note that I currently also have #263035 up

Apologies. Would it be more acceptable if I add you as a Co-Author and maintainers, or I could also delete this package in this PR.

**Comment by h7x4** [comment]
No worries :)

Let's use this PR, since it has more going for it. I'd be happy to co-maintain the package.

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (Cryolitia) [reply pkgs/by-name/fc/fcitx5-material-color/package.nix]: I copied this part from [archlinux cn PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/fcitx5-material-color/-/blob/main/PKGBUILD?ref_type=heads#L16), and I have tested stuff-in-one-directory but it doesn't work on my computer.

Screenshot 2023-10-26 09-49-39

Comment (h7x4) [reply pkgs/by-name/fc/fcitx5-material-color/package.nix]: I remember fighting with this package on arch linux like 2 years ago or something as well. I never managed to make it work, but I hoped it would be easier with NixOS.

Could you add a TODO checkbox at the top of the PR, so we remember to at least confirm it works once before merging?

File: pkgs/by-name/fc/fcitx5-nord/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-nord/package.nix:20]:
```suggestion
  installPhase = ''
    runHook preInstall
  
    mkdir -pv $out/share/fcitx5/themes/
    cp -rv Nord* $out/share/fcitx5/themes/
    
    runHook postInstall
  '';
```

File: pkgs/by-name/fc/fcitx5-nord/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-nord/package.nix:26]:
```suggestion
    maintainers = with maintainers; [ Cryolitia ];
    platforms = platforms.all;
```

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-material-color/package.nix:38]:
```suggestion
    maintainers = with maintainers; [ Cryolitia h7x4 ];
    platforms = platforms.all;
```

**Comment by h7x4** [comment]
I don't have time to test this stuff right now. I'll test it in the upcoming days if it hasn't been merged already

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (Cryolitia) [reply pkgs/by-name/fc/fcitx5-material-color/package.nix]: Screenshot 2023-10-26 10-10-22

「Stuff around into each separate directory」works on my computer, could we check the checkbox

File: pkgs/by-name/fc/fcitx5-nord/package.nix
Comment (h7x4) [inline pkgs/by-name/fc/fcitx5-nord/package.nix:22]: eval seems to complain about this whitespace

```suggestion

    mkdir -pv $out/share/fcitx5/themes/
    cp -rv Nord* $out/share/fcitx5/themes/

```

File: pkgs/by-name/fc/fcitx5-material-color/package.nix
Comment (h7x4) [reply pkgs/by-name/fc/fcitx5-material-color/package.nix]: Sure!

**Review by h7x4 — APPROVED** [review]
Verified the user id, seems good: https://api.github.com/user/23723294

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [ ] executables tested on x86_64-linux (maintainer tested, and it seems fine)
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

Result of `nixpkgs-review pr 263319` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fcitx5-material-color</li>
    <li>fcitx5-nord</li>
  </ul>
</details>

LGTM :+1:

**Review by NickCao — APPROVED** [review]

**Pull request merged**
